src/instruments/view/instrumentpaneltreemodel.cpp

```diff
@@ -146,6 +146,11 @@ void InstrumentPanelTreeModel::selectRow(const QModelIndex& rowIndex)
 
 void InstrumentPanelTreeModel::addInstruments()
 {
+    notation::INotationInteractionPtr interaction = context()->currentNotation()->interaction();
+    if (interaction->isTextEditingStarted()) {
+        return;
+    }
```

Comment:
I believe that we should check for the current `ShortcutContext` value (and extend and develop it), not add dependencies

`ShortcutContext GlobalContext::currentShortcutContext() const`

Reply:
Assigning a value like ```NotationTextEditing``` only works for shortcuts that need to work only when editing text. What we need here is the opposite of that? Not sure how I should go about implementing this.

Reply:
I am now writing documentation on shortcuts, and I found that, in fact, this system is broken. I ask you to pause work on this task for now. We (the in-house team) will fix the shortcut system, and then we will fix this issue

Reply:
Thank you for noticing this issue.

Reply:
No problem.